src/server/document-controller.js

```diff
@@ -198,6 +198,17 @@ class DocumentController {
     return res.send(result.length ? result[0].uniqueTags : []);
   }
 
+  async handlePostRegenerateDocumentsBatch(req, res) {
+    const { user } = req;
+    const batch = await this.documentService.createRegenerateDocumentsBatch(user);
+
+    if (!batch) {
+      throw new BadRequest('Another document regeneration is in progress');
```

Comment:
Why not do this the same way as in import batches and throw in the service?

I think that would be better, because `!batch` can mean also other things (a bug, `createRegenerateDocumentsBatch` having changed, who knows?)

Reply:
The current document-service doesn't use any httpErrors, only  the document-controller does. In my opinion is that this is the right way to do it, throwing http errors in a service makes it tied to the way it is exposed (http) to the outside and less reusable in other contexts. 

I know it's convenient to throw those errors from the service, but diminishes the use of the handlers in our controllers.

If we decide for this change I would move all errors from the handlers to the service.

Reply:
For me `!batch` does not directly translate into `BadRequest`, it just translates into batch not created, the reason of which only the service knows.

Reply:
I see both points. We are using http errors in other services, though, because they transport meaning. Returning `null` is a too magical "code" for "Another document regeneration is in progress". The only alternative would be to throw a custom error (but that would also introduce a new concept).

Reply:
I've mode the BadRequest handling to the service.